app/cloudinit/cloudinit.go

```diff
@@ -492,6 +492,84 @@ runcmd:
 
 var ubuntuTemplate = template.Must(template.New(oshelp.OSLinux).Funcs(funcs).Parse(ubuntuScript))
 
+const gitspacesUbuntuScript = `
```

Comment:
probably we should make it generic, and not make it gitspaces specific , something like a script add on, and then call it with gitspaces specific code

Reply:
We can make it generic, but we would still need to duplicate the script. The initial suggestion from CI team was to not touch their script

Reply:
I agree, either let's incorporate it into the main scripts if the delta is very less. Or otherwise optimise it to only install required tools for gitspaces

Reply:
updated the script, installing only required things now